File: backend/internal/model/company.go
package model

import "time"

// Company は法人情報（ノア動物病院、シングルトン）
type Company struct {
	ID                        uint64    `gorm:"primaryKey;autoIncrement"  json:"id"`
	Name                      string    `gorm:"not null"                  json:"name"`
	PostalCode                string    `gorm:"default:''"                json:"postal_code"`
	Address                   string    `gorm:"default:''"                json:"address"`
	PhoneNumber               string    `gorm:"default:''"                json:"phone_number"`
	FaxNumber                 string    `gorm:"default:''"                json:"fax_number"`
	Email                     string    `gorm:"default:''"                json:"email"`
	Website                   string    `gorm:"default:''"                json:"website"`
	DirectorName              string    `gorm:"default:''"                json:"director_name"`
	RegistrationNumber        string    `gorm:"default:''"                json:"registration_number"`
	InvoiceRegistrationNumber string    `gorm:"default:''"                json:"invoice_registration_number"`
	LogoURL                   string    `gorm:"default:''"                json:"logo_url"`
	CreatedAt                 time.Time `gorm:"autoCreateTime"            json:"created_at"`
	UpdatedAt                 time.Time `gorm:"autoUpdateTime"            json:"updated_at"`
}

func (Company) TableName() string { return "companies" }

File: frontend/src/features/master/api/company.ts
import { useQuery } from "@tanstack/react-query";
import { axios } from "@/lib/axios";
import { QUERY_STALE_TIMES, QUERY_GC_TIMES } from "@/lib/react-query";
import type { Company as ModelCompany } from "@/types/generated/models";

// ─────────────────────────────────────────────────
// Transform
// ─────────────────────────────────────────────────

function transformCompany(data: ModelCompany) {
  return {
    id: data.id,
    name: data.name,
    postalCode: data.postal_code,
    address: data.address,
    phoneNumber: data.phone_number,
    faxNumber: data.fax_number,
    email: data.email,
    website: data.website,
    directorName: data.director_name,
    registrationNumber: data.registration_number,
    invoiceRegistrationNumber: data.invoice_registration_number,
    logoUrl: data.logo_url,
    createdAt: data.created_at,
    updatedAt: data.updated_at,
  };
}

type Company = ReturnType<typeof transformCompany>;

// ─────────────────────────────────────────────────
// Query keys
// ─────────────────────────────────────────────────

const COMPANY_QUERY_KEY = ["masters", "company"] as const;

// ─────────────────────────────────────────────────
// API functions
// ─────────────────────────────────────────────────

async function getCompany(): Promise<Company> {
  const { data } = await axios.get<ModelCompany>("/v1/company");
  return transformCompany(data);
}

// ─────────────────────────────────────────────────
// TanStack Query hooks
// ─────────────────────────────────────────────────

export function useGetCompany() {
  return useQuery({
    queryKey: COMPANY_QUERY_KEY,
    queryFn: getCompany,
    staleTime: QUERY_STALE_TIMES.STATIC,
    gcTime: QUERY_GC_TIMES.LONG,
  });
}
